core/src/main/java/com/linecorp/armeria/server/logging/structured/StructuredLogWriter.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.logging.structured;
+
+import com.linecorp.armeria.common.RequestContext;
+import com.linecorp.armeria.common.logging.ResponseLog;
+
+/**
+ * An interface that a structured log writer should implement.
+ * @param <L> type of the structured log representation
+ */
+@FunctionalInterface
+public interface StructuredLogWriter<L> {
```

Comment:
In some cases L could be not a subclass of StructuredLog.
For example, if a user decided to use Protocol buffer or Thrift as a serialization format of structured log(as I mentioned in the PR description), L might be a subclass of `MessageLite` or `TBase`. They aren't a subclass of `StructuredLog` so that type boundary doesn't work.